internal/cli/template_list.go

```diff
@@ -0,0 +1,95 @@
+package cli
+
+import (
+	"strings"
+
+	"github.com/olekukonko/tablewriter"
+	"github.com/posener/complete"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+)
+
+type ProjectTemplateListCommand struct {
+	*baseCommand
+}
+
+func (c *ProjectTemplateListCommand) Run(args []string) int {
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(c.Flags()),
+		WithNoConfig(),
+	); err != nil {
+		return 1
+	}
+	ctx := c.Ctx
+
+	out, _, err := c.ui.OutputWriters()
+	if err != nil {
+		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+		return 1
+	}
+
+	ptr, err := c.project.Client().ListProjectTemplates(ctx, &pb.ListProjectTemplatesRequest{})
+	if err != nil {
+		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+		return 1
+	}
+	templates := ptr.ProjectTemplates
+
+	if len(templates) == 0 {
+		c.ui.Output("No project templates found.")
```

Comment:
I think other CLI commands in Waypoint do this, but maybe we just output nothing in the 0 case?